meson.build

```diff
@@ -0,0 +1,150 @@
+project('GMime',
+	['c'],
+	license: 'GPLv2',
+    version: '3.2.13',
+    meson_version: '>=1.0.0',
+    default_options : [
+    ])
+
+gmime_version = meson.project_version()
+ver_arr = gmime_version.split('.')
+
+GMIME_MAJOR_VERSION=ver_arr[0].to_int()
+GMIME_MINOR_VERSION=ver_arr[1].to_int()
+GMIME_MICRO_VERSION=ver_arr[2].to_int()
+GMIME_BINARY_AGE= 100 * GMIME_MINOR_VERSION + GMIME_MICRO_VERSION
+GMIME_INTERFACE_AGE=0
+
+gmime_package = 'gmime-@0@.0'.format(GMIME_MAJOR_VERSION)
+
+add_global_arguments('-DHAVE_CONFIG_H=1', language : 'c')
+
+iconv = dependency('iconv', required : false, method: get_option('iconv'))
+libidn2 = dependency('libidn2', version: '>=2.0.0', required : false)
+gpgme = dependency('gpgme', version: '>=1.6.0', required: get_option('crypto'))
+glib =  dependency('glib-2.0', version: '>=2.68')
+gobject = dependency('gobject-2.0')
+deps = [
+  dependency('zlib', version: '>=1.2.5.2'),
+  glib,
+  gpgme,
+  gobject,
+  dependency('gio-2.0'),
+  libidn2,
+  iconv,
+]
+
+gmime_prefix = get_option('prefix')
+gmime_datadir = join_paths(gmime_prefix, get_option('datadir'))
+
+cc = meson.get_compiler('c')
+
+conf_data = configuration_data()
+conf_data.set_quoted('GMIME_VERSION', meson.project_version())
+
+LIBS = ['glib-2.0', 'gio-2.0', 'gobject-2.0' ]
+EXTRA_LIBS = ['zlib']
+
+if cc.has_header('getopt.h')
+  if cc.has_function('getopt_long')
+    conf_data.set('HAVE_GETOPT_H', 1)
+  endif
+endif
+
+if gpgme.found()
+  conf_data.set('ENABLE_CRYPTO', 1)
+  EXTRA_LIBS += 'gpgme'
+  EXTRA_LIBS += 'gpg-error'
+endif
+
+if libidn2.found()
+  conf_data.set('LIBIDN', 1)
+  EXTRA_LIBS += 'libidn2'
+endif
+
+
+iconv_detect = '''
+#define CONFIGURE_IN
+#include "iconv-detect.c"
+'''
+
+if get_option('have_iconv_detect_h')
+  conf_data.set('HAVE_ICONV_DETECT_H', 1)
+else
+  # not cross-compiler secure
+  root = include_directories('.')
+  has_detect = cc.run(iconv_detect, name: 'iconv-detect', 
+    include_directories: root).returncode() == 0
+  if has_detect
+    conf_data.set('HAVE_ICONV_DETECT_H', 1)
+  endif
+endif
```

Comment:
Meson can absolutely handle this with cross compilation, you just have to define an exe_wrapper (usually qemu-user). If you don't, meson errors out with "Can not run test applications in this cross environment." which... I suppose could be enhanced to mention exe_wrapper, hmm.

But this is a pretty awkward way to do things here. You're running a test program in the source directory and emitting unclean output into the source directory instead of the build directory -- this violates Meson's guarantee of out-of-source build directories, potentially messing up situations such as read-only mounted source directories or having multiple build directories, one native one cross compiled, which need *different* generated headers.

Reply:
Yeah. Adding an error if we can't run. 

In the long run I think it might be better to not pollute the system. The reason for me to "port the awkwardness" over is to not disrupt the autotools build which can require this file to exist.

Reply:
As someone who cross-compiles daily, remember to include a fallback for cross and can't run by checking for a property in the cross file.

Reply:
To be fair, this code only runs in the else branch of a `get_option()`...